### PR TITLE
Create PR/branch/ami jobs for ovn-kubernetes project

### PR DESCRIPTION
**Pull request by rajatchopra**
Signed-off-by: Rajat Chopra <rchopra@redhat.com>

@stevekuznetsov PTAL
I still don't have a great idea about what is going on here. Please review this attempt.

### CODE REVIEW

File: sjb/config/common/test_cases/ovn-kubernetes.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+---
```
Comment (stevekuznetsov) [inline sjb/config/common/test_cases/ovn-kubernetes.yml]: How is this different from e.g. the CRI-O common base?

File: sjb/config/test_cases/ami_build_origin_int_rhel_ovn_kubernetes.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/ami_build_origin_int_rhel_ovn_kubernetes.yml]: How much time does this save? Handling a separate set of AMIs is a big maintenance burden

File: sjb/config/test_cases/test_pull_request_ovn_kubernetes_rhel.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - type: "forward_parameters"
+    parameters:
+      - PULL_NUMBER
+      - PULL_BASE_SHA
+      - PULL_PULL_SHA
+  - type: "script"
+    title: "clone ovn-kubernetes and check out the correct refs"
+    script: |-
+      cd /go/src/github.com/openvswitch/ovn-kubernetes
+      git fetch origin
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_ovn_kubernetes_rhel.yml]: You don't need to name your refs:

```sh
repo="https://github.com/openvswitch/ovn-kubernetes.git"
git init
git fetch "${repo}" "${PULL_BASE_REF}"
git checkout "${PULL_BASE_SHA:-'FETCH_HEAD'}"
git fetch "${repo}" "pull/${PULL_NUMBER}/head"
git merge "${PULL_PULL_SHA:-'FETCH_HEAD'}"
```

But if you want to handle batch jobs this gets more complicated...

File: sjb/config/test_cases/test_pull_request_ovn_kubernetes_rhel.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      git checkout master
+      git branch -D target || true
+      git branch -D pr || true
+      git branch target "${PULL_BASE_SHA}"
+      git checkout target
+      git fetch origin "pull/${PULL_NUMBER}/head:pr"
+      git merge "${PULL_PULL_SHA}"
+  - type: "script"
+    title: "run the cri-o integration tests"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_ovn_kubernetes_rhel.yml]: typo?

File: sjb/config/common/test_cases/ovn-kubernetes.yml (hunk 1 above)
Comment (rajatchopra) [reply sjb/config/common/test_cases/ovn-kubernetes.yml]: Just as much as artifacts are different. And we don't care about some journal files to be captured. Later on, I expect some local files to be gathered.
It won't harm to use crio at all, just would be surplus processing. Should we scrap this file to save clutter in the repo? Pretty sure we might need some customizations later on though.

File: sjb/config/test_cases/test_pull_request_ovn_kubernetes_rhel.yml (hunk 3 above)
Comment (rajatchopra) [reply sjb/config/test_cases/test_pull_request_ovn_kubernetes_rhel.yml]: Just copied the stuff. I doubt if we need batch jobs.

File: sjb/config/test_cases/test_pull_request_ovn_kubernetes_rhel.yml (hunk 4 above)
Comment (rajatchopra) [reply sjb/config/test_cases/test_pull_request_ovn_kubernetes_rhel.yml]: Fixed

File: sjb/config/test_cases/ami_build_origin_int_rhel_ovn_kubernetes.yml (hunk 2 above)
Comment (rajatchopra) [reply sjb/config/test_cases/ami_build_origin_int_rhel_ovn_kubernetes.yml]: As much as:
1. Clone openshift origin source
2. Compile openshift origin
3. Build dind images

Can we re-use some AMI that already has openshift origin source in it?

File: sjb/config/common/test_cases/ovn-kubernetes.yml (hunk 1 above)
Comment (stevekuznetsov) [reply sjb/config/common/test_cases/ovn-kubernetes.yml]: If it's possible, you could inherit from this one and just provide `overrides:` for the artifacts and debug journal files.

File: sjb/config/test_cases/ami_build_origin_int_rhel_ovn_kubernetes.yml (hunk 2 above)
Comment (stevekuznetsov) [reply sjb/config/test_cases/ami_build_origin_int_rhel_ovn_kubernetes.yml]: The `build` stage of Origin AMIs has latest Origin master built and compiled. Do you want to run against latest? Latest stable?

Comment (stevekuznetsov) [reply sjb/config/test_cases/ami_build_origin_int_rhel_ovn_kubernetes.yml]: The `build` stage of Origin AMIs has latest Origin master built and compiled. Do you want to run against latest? Latest stable?

**Comment by rajatchopra** [comment]
Closing this in favour of https://github.com/openshift/aos-cd-jobs/pull/1119

**Pull request closed without merging**